src/main/java/org/jitsi/jicofo/JitsiMeetConference.java

```diff
@@ -87,6 +88,12 @@
      */
     void setStartMuted(boolean[] startMuted);
 
+    /**
+     * @return the list of all Jibri sessions(of any type) used by this
+     * conference.
+     */
+    List<JibriSession> getJibriSessions();
```

Comment:
Instead of adding code to reach in and extract all Jibri sessions, can we just have the code push stats up to somewhere which would cache the counts and then we can query that cache for the stats?

Reply:
do you mean to track the events when sessions starts etc. ? I'd rather not do that as it's more error prone

Reply:
Yes I mean the code that starts a session fires a stat up that a session has started, etc.   What makes it more error prone?

Reply:
That listeners must always be in place and the events have to be fired correctly, missing one event for transition to ended or failed will cause a leak. Querying the current state is also easier in my opinion. Why do you think events are better? Is it the list of all sessions that bothers you?

Reply:
I was assuming we'd have a single static listener which always existed and would capture these and also be query-able for stats.  But yeah, I just thought it was unfortunate to have to plumb a call all the way down to get the list of sessions.

Reply:
It is a bit unfortunate, but it's either that global query or global events.

Reply:
Maybe I'm misreading this, but I thought the stats were already global...the `eventAdmin` calls?

Reply:
I also think that counting events creates a second source of truth. Why not just generate the stats based on the real current state.

Reply:
Just because it breaks encapsulation more to reach down and grab them.

Reply:
That may also require locking, etc. to grab things which can conflict with code execution.  There's still a chance we just miss the start or end of a session.

Reply:
if instead of returning the entire list I've passed a selector? is that better? then it's api just for querying and does not expose the whole set?

Reply:
I'm also confused because we already had a `JibriStats` class which was tracking failures...why can't we just add these to that using the same method? Why push some stats up one way, and query for others?

Reply:
because failures are not stored in any state, so it  creates a new state that stores them

Reply:
So I guess I'm wondering: if we already have a mechanism for pushing Jibri-related events for the purposes of stat tracking, why not re-use it?

Reply:
Thinking about it more, I do see how maybe we'd need to include a jibri ID or some common identifier for state  to track, for example, when a jibri goes from "available" to "busy" or something.  I do see how that's a bit more complicated than tracking error counts, and iterating over the current state makes it easier.  I don't know if it would be easy to deal with that (with a jibri id or whatever)

Reply:
there are no events really for tracking  JibriSession's current status, so I'd have to add global events, then create a tracking object to update the current state and then count them anyway

Reply:
yes, what you wrote above, it is more work and it creates another source of truth

Reply:
I don't agree with the source of truth argument, because I don't see why pushing information up wouldn't be reliable, but I see that it can be complex to track the current state of sessions (primarily tracking which state they _left_).

Reply:
my proposal is to change the getter of all session to a query method  for  counting them

Reply:
If you mean adding something like a `getJibriSessionStats` method which returned some stats object or a `JSONObject`, then yeah I do think that would be better.

Reply:
ok